File: Practice/src/introduction_coding_tests/LengthOfOverlappingSegments.js
function solution(lines) {
    var answer = 0;
    const arr = Array(201).fill(0);
    for (let [x, y] of lines) {
        for (let i = x + 100; i < y + 100; i++) {
            arr[i]++;
        }
    }
    for (let i of arr) {
        if (i > 1) answer++;
    }
    return answer;
}

File: Practice/src/introduction_coding_tests/LengthOfOverlappingSegments.java
package introduction_coding_tests;

public class LengthOfOverlappingSegments {
//	겹치는 선분의 길이
	public int solution(int[][] lines) {
		int answer = 0;
		int[] arr = new int[201];
		for (int i = 0; i < 3; i++) {
			for (int j = lines[i][0] + 100; j < lines[i][1] + 100; j++) {
				arr[j]++;
			}
		}
		for (int i : arr) {
			if (i > 1)
				answer++;
		}
		return answer;
	}
}